lib/XML/LibXML/Document.pm

```diff
@@ -341,3 +374,37 @@ method new-cdata-block(Str $cdata) {
     nqp::bindattr(nqp::decont($node), xmlNode, '$!doc', nqp::decont(self));
     $node
 }
+
+method createTextNode(Str $content) {
+    my $newText = self.new-text($content);
+    my $docfrag = self.new-doc-fragment();
+    xmlAddChild(
+        nativecast(xmlNodePtr, $docfrag), 
+        nativecast(xmlNodePtr, $newText)
+    );
+
+    $newText;
+}
+
+method setDocumentElement($e) {
+    my $elem = nativecast(xmlNode, $e);
+
+    if $elem.type != XML_ELEMENT_NODE {
+        die "setDocumentElement: ELEMENT node required";
+        # cw: To properly handle .resume in CATCH {}
+        return;
+    }
+
+    domImportNode(self, $elem, 1, 1);
+    my $oelem = xmlDocGetRootElement(self);
+    if (!$oelem.defined || !$oelem._private.defined) {
+        xmlDocSetRootElement(self, $elem);
+    } else {
```

Comment:
LOL! First time I've heard that term. That's an old coding style I adopted which I am happy to discard. Means a lot of changes though. I will knock them out as I find them.